specification/search/data-plane/Azure.Search/preview/2025-05-01-preview/searchservice.json

```diff
@@ -0,0 +1,14715 @@
+                "description": "The role of the message sender."
+              },
+              "content": {
+                "type": "string",
+                "description": "The content of the message. Mutually exclusive with multimodalContent."
+              },
+              "multimodalContent": {
+                "type": "array",
+                "items": {
+                  "type": "object",
+                  "properties": {
+                    "type": {
+                      "type": "string",
+                      "description": "The type of the message content."
+                    },
+                    "text": {
+                      "type": "string",
+                      "description": "The text of a string message content."
+                    },
+                    "image": {
+                      "type": "object",
+                      "properties": {
+                        "url": {
+                          "type": "string",
+                          "description": "The url of the image"
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            },
+            "required": [ "role" ]
+          }
+        },
+        "includeReferenceSourceData": {
+          "type": "boolean",
+          "description": "Whether to include the source data in the response references."
+        }
+      },
+      "description": "The input contract for the retrieval request."
+    },
+    "KnowledgeAgentRetrievalResponse": {
+      "type": "object",
+      "properties": {
+        "response": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "role": {
+                "type": "string",
+                "description": "The role of the tool response."
+              },
+              "content": {
+                "type": "array",
+                "items": {
+                  "type": "object",
+                  "properties": {
+                    "type": {
+                      "type": "string",
+                      "description": "The type of the message content."
+                    },
+                    "text": {
+                      "type": "string",
+                      "description": "The text of a string message content."
+                    },
+                    "image": {
+                      "type": "object",
+                      "properties": {
+                        "url": {
+                          "type": "string",
+                          "description": "The url of the image"
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            },
+            "required": [ "role", "content" ]
+          }
+        },
+        "activity": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/KnowledgeAgentActivityRecord"
```

Comment:
Will the activity records here be all of one type or could they be a mix of all the activity record types? Ex, are these all Search activity records, or could you have a mix of Search, query planning, etc?

If it's the latter, how would someone consuming this field know which kind of activity record to deserialize into? There's no way to determine that based on the Swagger definitions here.